### PR TITLE
Feat/integration tests

### PR DESCRIPTION
**Pull request by paasovaara**
First unit and integration tests. Usage instructions in Readme.

also removed reference to junit5, let's go with junit4.

Contains rollback for jackson-xml ClassNotFound exception

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -138,6 +139,7 @@
                 <artifactId>maven-assembly-plugin</artifactId>
                 <executions>
                     <execution>
+                        <id>make-assembly</id> <!-- this is used for inheritance merges -->
```
Comment (haphut) [inline pom.xml]: Is the comment relevant for our project?

**Comment by haphut** [comment]
Merging `master` into the feature branch looks messy in commit f0543d4924994fdfaf2a198b5f61e542747e77f7 . Rebase please.

**Comment by haphut** [comment]
Please harmonize the git commit message subject lines. Usual convention is to write them so that they finish the sentence "If applied, this commit will <your subject line here>" but starting with a capital letter, the so-called FIXME format.

**Comment by haphut** [comment]
I get the following warnings from the integration tests:
```
[WARNING] Configuration options: 'appendAssemblyId' is set to false, and 'classifier' is missing.
Instead of attaching the assembly file: /home/haphut/prog/hsldevcom/transitdata/transitdata-common/target/transitdata-common-0.1.2.jar, it will become the file for main project artifact.
NOTE: If multiple descriptors or descriptor-formats are provided for this project, the value of this file will be non-deterministic!
[WARNING] Replacing pre-existing project main-artifact file: /home/haphut/prog/hsldevcom/transitdata/transitdata-common/target/transitdata-common-0.1.2.jar
with assembly file: /home/haphut/prog/hsldevcom/transitdata/transitdata-common/target/transitdata-common-0.1.2.jar
```

**Comment by haphut** [comment]
I tried to run the integration tests wherever I dared. I got the error: `Could not start container`.

**Comment by paasovaara** [comment]
> Merging `master` into the feature branch looks messy in commit [f0543d4](https://github.com/HSLdevcom/transitdata-common/commit/f0543d4924994fdfaf2a198b5f61e542747e77f7) . Rebase please.



> I get the following warnings from the integration tests:
> 
> ```
> [WARNING] Configuration options: 'appendAssemblyId' is set to false, and 'classifier' is missing.
> Instead of attaching the assembly file: /PATH/TO/REPO/target/transitdata-common-0.1.2.jar, it will become the file for main project artifact.
> NOTE: If multiple descriptors or descriptor-formats are provided for this project, the value of this file will be non-deterministic!
> [WARNING] Replacing pre-existing project main-artifact file: /PATH/TO/REPO/target/transitdata-common-0.1.2.jar
> with assembly file: /PATH/TO/REPO/target/transitdata-common-0.1.2.jar
> ```

This should not come from integration tests, it's from the actual build. And it seems to be a known maven-assembly-plugin behaviour, could not find an easy way to suppress the warning (attempts only create new different warnings) https://issues.apache.org/jira/browse/MASSEMBLY-824

**Comment by paasovaara** [comment]
> I tried to run the integration tests wherever I dared. I got the error: `Could not start container`.

@haphut any more info about this one? I assume you have Docker installed? Do you already have some other docker container running in the ports that our tests are using, fex Redis? If this is the reason then we should just bind to another port during the launch.. however I think the testcontainers already does that....?

**Comment by paasovaara** [comment]
> Merging `master` into the feature branch looks messy in commit [f0543d4](https://github.com/HSLdevcom/transitdata-common/commit/f0543d4924994fdfaf2a198b5f61e542747e77f7) . Rebase please.

👍 let's use rebase from now on. usually I try to avoid it when there's a chance that people collaborate on branches, it has caused a mess sometimes. And agree that the commit message for this one was particulary bad, it was my mistake. usually I just use the default merge commit msg

**Comment by paasovaara** [comment]
Integration tests work in travis.

**Pull request merged**